File: 883-car-fleet/car-fleet.cpp
class Solution {
public:
    int carFleet(int target, vector<int>& position, vector<int>& speed) {
        map<int,int> m;
        for(int i=0; i<position.size(); i++){
            m[-position[i]] = speed[i];
        }
        int res = 0; double cur = 0;
        for(auto ele: m){
            double time = (double)(target+ele.first)/ele.second;
            if(time > cur){
                res++;
                cur = time;
            }
        }

        return res;
    }
};

File: 883-car-fleet/car-fleet.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        arr = [(position[i], speed[i]) for i in range(len(position))]
        arr.sort()
        res = 0
        curr_time = 0
        for ele in range(len(position)-1, -1, -1):
            time = (target - arr[ele][0]) / arr[ele][1]
            if time > curr_time:
                res += 1
                curr_time = time
        
        return res